tests/attestation_cert_apis/host/host.cpp

```diff
@@ -92,16 +94,28 @@ void run_test(oe_enclave_t* enclave, int test_type)
 
     {
         // for testing purpose, output the whole cer in DER format
-        char filename[80];
+        char filename[FILENAME_LEN];
         FILE* file = NULL;
 
+#ifdef _WIN32
+        sprintf_s(
+            filename,
+            FILENAME_LEN,
+            "./cert_%s.der",
+            test_type == TEST_RSA_KEY ? "rsa" : "ec");
+#else
         sprintf(
             filename,
             "./cert_%s.der",
             test_type == TEST_RSA_KEY ? "rsa" : "ec");
+#endif
         OE_TRACE_INFO(
             "Host: Log quote embedded certificate to file: [%s]\n", filename);
+#ifdef _WIN32
+        fopen_s(&file, filename, "wb");
+#else
         file = fopen(filename, "wb");
```

Comment:
same, just use fopen_s always

Reply:
Tried and found that fopen_s() does a good job anywhere except in another file, tests/crypto/read_file.c. It does not help with <stdio.h> explicitly included. I'm wondering: is there anything particular with this file? Here's an error message:
-----
[1/2] Building C object tests\crypto\enclave\enc\CMakeFiles\cryptoenc.dir\__\__\read_file.c.obj
FAILED: tests/crypto/enclave/enc/CMakeFiles/cryptoenc.dir/__/__/read_file.c.obj 
"C:/Program Files/Git/bin/bash.exe" "C:/Users/ryanhsu20/ver2/openenclave/scripts/clangw" "-DMBEDTLS_CONFIG_FILE="<config.h>" -DOE_API_VERSION=2 -DOE_BUILD_ENCLAVE -DOE_LINK_SGX_DCAP_QL -Itests\crypto\enclave\enc -I..\..\3rdparty\mbedtls\mbedtls\include -I3rdparty\mbedtls -I..\..\include -Ioutput\include\openenclave\libc /DWIN32 /D_WINDOWS /wd4566 /wd4200 /MD /O2 /Ob2 /DNDEBUG   /WX /W3 -Wall -Werror -Wpointer-arith -Wconversion -Wextra -Wno-missing-field-initializers -fno-strict-aliasing -mxsave -fno-builtin-malloc -fno-builtin-calloc -fno-builtin -mllvm -x86-speculative-load-hardening -m64 -fPIE -nostdinc -fstack-protector-strong -fvisibility=hidden -fno-omit-frame-pointer -ffunction-sections -fdata-sections -ftls-model=local-exec /showIncludes -o tests\crypto\enclave\enc\CMakeFiles\cryptoenc.dir\__\__\read_file.c.obj -c ..\..\tests\crypto\read_file.c"
..\..\tests\crypto\read_file.c:13:5: error: implicit declaration of function 'fopen_s' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    fopen_s(&cfp, filename, "rb");
    ^
..\..\tests\crypto\read_file.c:13:5: note: did you mean 'fopen'?
output\include\openenclave\libc\stdio.h:64:7: note: 'fopen' declared here

Reply:
ah, tests/crypto runs in both enclave and host, so the enclave version will fail to build. Go ahead and just use an ifdef here.

Reply:
Got it, thanks!